### PR TITLE
Fix cache race

### PR DESCRIPTION
**Pull request by aschmahmann**

### CODE REVIEW

File: resolver.go [hunk 1]
```diff
@@ -13,7 +13,7 @@ import (
 )
 
 type Resolver struct {
-	sync.RWMutex
+	mx sync.Mutex
```
Comment (aschmahmann) [inline resolver.go:16]: Figured exporting the Lock/Unlock functions here was unintentional and a bad idea for a consumer to use.

File: resolver.go [hunk 2]
```diff
@@ -101,8 +101,8 @@ func (r *Resolver) LookupTXT(ctx context.Context, domain string) ([]string, erro
 }
 
 func (r *Resolver) getCachedIPAddr(domain string) ([]net.IPAddr, bool) {
-	r.RLock()
-	defer r.RUnlock()
+	r.mx.Lock()
```
Comment (aschmahmann) [inline resolver.go:104]: Switching away from RW mutexes may slow things down here. Given that we're caching against what would be a network request waiting on a mutex doesn't feel too bad, and if this becomes an issue we can always come back and optimize it.

File: resolver.go (hunk 1 above)
Comment (vyzo) [reply resolver.go:16]: ah yes, it was just programmatic convenience!

File: resolver.go (hunk 2 above)
Comment (vyzo) [reply resolver.go:104]: let's add a follow up issue? It shouldn't be hard to do, just queue the entries for cleanup and do that with write lock.

**Review by vyzo — APPROVED** [review]

File: resolver.go (hunk 2 above)
Comment (aschmahmann) [reply resolver.go:104]: https://github.com/libp2p/go-doh-resolver/issues/5

**Pull request merged**